README.md

```diff
@@ -576,20 +576,259 @@ Permit or deny Fragments for ACL. Valid values are 'permit-all' and 'deny-all'
 
-| Minimum Requirements | N9k | N30xx | N31xx | N56xx | N6k | N7k |
-|----------------------|:---:|:-----:|:-----:|:-----:|:---:|:---:|
-| OS Image | 7.0(3)I2(1) | 7.0(3)I2(1) | 7.0(3)I2(1) | 7.3(0)N1(1) | 7.3(0)N1(1) | 7.3(0)D1(1) |
-| Puppet Module | 1.2.0 | 1.2.0 | 1.2.0 | unsupported | unsupported | unsupported |
+| Platform | OS Minimum Version | Module Minimum Version |
+|----------|:------------------:|:----------------------:|
+| N9k      | 7.0(3)I2(1)        | 1.2.0                  |
+| N30xx    | 7.0(3)I2(1)        | 1.2.0                  |
+| N31xx    | 7.0(3)I2(1)        | 1.2.0                  |
+| N56xx    | unsupported        | unsupported            |
+| N6k      | unsupported        | unsupported            |
+| N7k      | unsupported        | unsupported            |
+| IOS XR   | unsupported        | unsupported            |
+
+#### <a name="cisco_ace-caveats">Caveats</a>
+
+| Property | Caveat Description |
+|:--------|:-------------|
+| `http_method` | Not supported on N56xx, N6k, N7k |
+| `packet_length` | Not supported on N56xx, N6k |
+| `redirect` | Not supported on N56xx, N6k, N7k |
+| `time_range` | Not supported on N56xx, N6k |
+| `ttl` | Not supported on N56xx, N6k, N7k |
+| `tcp_option_length` | Not supported on N56xx, N6k, N7k |
+
+#### Example Usage
+
+```puppet
+cisco_ace { 'ipv4 my_acl 42':
+  ensure              => 'present',
+  remark              => 'East Branch',
+  action              => 'permit',
+  proto               => 'tcp',
+  src_addr            => '10.0.0.0/8',
+  src_port            => 'eq 40',
+  dst_addr            => 'any',
+  dst_port            => 'neq 80',
+
+  dscp                => 'af11',
+  established         => 'true',
+  http_method         => 'post',
+  log                 => 'true',
+  packet_length       => 'range 512 1024'
+  precedence          => 'flash',
+  redirect            => 'Ethernet1/2,Port-Channel42',
+  tcp_flags           => 'ack psh',
+  tcp_option_length   => '36',
+  time_range          => 'my_time_range',
+  ttl                 => '128',
+```
 
 #### Parameters
 
-##### `ensure`
-Determines whether the config should be present or not on the device. Valid values are 'present' and 'absent'.
+| Example Parameter Usage
+|:--
+| `cisco_ace { '<afi> <acl_name> <seqno>':`
+| `cisco_ace { 'ipv4 my_acl 42':`
 
 ##### `afi`
 Address Family Identifier (AFI). Required. Valid values are ipv4 and ipv6.
 
-*TBD: Add property definitions*
+##### `acl_name`
+Access Control List (ACL) name. Required. Valid values are type String.
+
+##### `seqno`
+Access Control Entry (ACE) Sequence Number. Required. Valid values are type Integer.
+
+##### `ensure`
+Determines whether the config should be present or not on the device. Valid values are 'present' and 'absent'.
+
+#### Properties
+
+##### `action`
+The action to perform with this ACE. Valid values are keywords `permit` or `deny`.
+
+| Example
+|:--
+| `action => 'permit'`
```

Comment:
True, this format isn't there for other providers but one of the comments we got back from the TMEs was that they had difficulty figuring out how to use some of these properties; ie. we have examples in the type definitions but you have to look up each type file or do puppet describe to see that; so this was an attempt to get this documentation moving in a direction that included example usage.

Do we need examples for every property? Maybe not. I added them for completeness. I'd prefer to keep them but we can remove them if you feel strongly about it.